sharc/propagation/propagation_input_file.py

```diff
@@ -35,28 +36,76 @@ def __init__(self, input_folder:str):
         self.files = []
         self.path_loss = dict()
         
-        # Find where data begins
-        begin_data_line = 12
-        
         # Loop through all the txt files in the folder
         for file in glob(input_folder + "\\*.txt"):
             
             self.files.append(file)
             
             with open(file) as f:
                 
-                # Treat file header
-                head_raw = [next(f) for x in range(begin_data_line)]
-                head = PathLossHeader(head_raw[0].split()[1][1:-1],
-                                      [float(x) for x in head_raw[1].split()[1:]],
-                                      float(head_raw[2].split()[1]),
-                                      head_raw[3].split()[1:],
-                                      head_raw[4].split()[1],
-                                      [float(x) for x in head_raw[5].split()[1:]],
-                                      [float(x) for x in head_raw[6].split()[1:]],
-                                      float(head_raw[7].split()[1]),
-                                      float(head_raw[8].split()[1]),
-                                      float(head_raw[9].split()[1]))
+                # Find where data begins and extract header
+                head_raw = ""
+                line = next(f)
+                while "BEGIN_DATA" not in line:
+                    head_raw = head_raw + line
+                    line = next(f)
+                    
+                # Find tokens in file header
+                head_list = head_raw.split() + 4*[np.nan]
+                param_list = ["ANTENNA",
+                              "LOCATION",
+                              "FREQUENCY",
+                              "POWER",
+                              "ANTENNATYPE",
+                              "LOWER_LEFT",
+                              "UPPER_RIGHT",
+                              "HEIGHT",
+                              "RESOLUTION",
+                              "RECEIVER_GAIN"]
+                essential_params = ["ANTENNA",
+                                    "LOWER_LEFT",
+                                    "UPPER_RIGHT",
+                                    "RESOLUTION",
+                                    "RECEIVER_GAIN"]
+                idx_list = []
+                for param in param_list:
+                    try:
+                        idx_list.append(head_list.index(param))
+                    except ValueError as err:
+                        if param in essential_params:
+                            sys.stderr.write(param + 
+                                         " parameter not in path loss file " +
+                                         file + "\n")
+                            sys.exit(1)
+                        else:
+                            idx_list.append(-4)
+                    
+                # Get parameter values
```

Comment:
The header parsing is heavily dependent on list indexing. This is very confusing and lacks performance. A better approach should be analysed.